pkg/hostagent/snats.go

```diff
@@ -339,6 +352,50 @@ func (agent *HostAgent) snaGlobalInfoChanged(snatobj interface{}, logger *logrus
 		}
 	}
 	agent.log.Debug("Snat Gobal Obj Map: ", agent.opflexSnatGlobalInfos)
+
+	snatFileName := SnatService + ".service"
+	filePath := filepath.Join(agent.config.OpFlexServiceDir, snatFileName)
+	file_exists := fileExists(filePath)
+
+	if len(agent.opflexSnatGlobalInfos) > 0 {
+		// if more than one global infos, create snat ext file
+		as := &opflexService{
+			Uuid: SnatService,
+			DomainPolicySpace: agent.config.AciVrfTenant,
+			DomainName: agent.config.AciVrf,
+			ServiceMode: "loadbalancer",
+			ServiceMappings: make([]opflexServiceMapping, 0),
+			InterfaceName: agent.config.UplinkIface,
+			InterfaceVlan: uint16(agent.config.ServiceVlan),
+			ServiceMac: agent.serviceEp.Mac,
+			InterfaceIp: agent.serviceEp.Ipv4.String(),
+		}
+		sm := &opflexServiceMapping{
+			Conntrack:    true,
+		}
+		as.ServiceMappings = append(as.ServiceMappings, *sm)
+		agent.opflexServices[SnatService] = as
+		if !file_exists {
```

Comment:
Not a big deal, but would you not be doing this check immediately after the L360 check? That way you don't do L362 to L377 processing if the file already exists.

Reply:
@apoorva11029 for the offline explanation as to why its done this way. Basically the reason seems to be that we want the cache to be populated correctly and it wont be if the file is present (and the check is moved t